eclipse-collections-code-generator/src/main/resources/test/map/mutable/unmodifiablePrimitivePrimitiveMapValuesTest.stg

```diff
@@ -88,13 +88,26 @@ public class Unmodifiable<name1><name2>MapValuesTest extends AbstractMutable<nam
         <name2>Iterator iterator = bag.<type2>Iterator();
         for (int i = 0; i \< 4; i++)
         {
-            Assert.assertTrue(iterator.hasNext());
-            Assert.assertTrue(list.remove(iterator.next()));
+            Assertions.assertTrue(iterator.hasNext());
+            Assertions.assertTrue(list.remove(iterator.next()));
         }
         Verify.assertEmpty(list);
-        Assert.assertFalse(iterator.hasNext());
+        Assertions.assertFalse(iterator.hasNext());
 
-        Assert.assertThrows(NoSuchElementException.class, iterator::next);
+        Assertions.assertThrows(NoSuchElementException.class, iterator::next);
+    }
+
+    @Override
+    @Test
+    public void <type2>Iterator_throws_non_empty_collection()
```

Comment:
@motlin that's an example of the **Override** that was necessary because the implementation that's inherited from the parent doesn't work in all subclasses.

It was passing before as it was expecting NoSuchElementException but this exception was coming from the method's super.testMethod() call that was failing, luckily with the correct exception, but nothing after the super.testMethod() call was actually executed. 

This became apparent only after moving to the new junit5 assertion style.

Reply:
We recently landed a few PRs with the goal of paving the way toward JUnit 5, like this one https://github.com/eclipse/eclipse-collections/pull/1555. In it, you can see that we got rid of some usages of `@Test(expected = ....class)` in templates. I thought we had got rid of all of them, but I see that we still haven't.

I'm having a tough time answering my own questions here because the diff is so big that GitHub collapses all the diffs. Is the test in the superclass changed o not use `expected`? I'd recommend making that refactoring in its own PR similar to the linked PR and then rebasing this on top.